plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryArrowToPageConverter.java

```diff
@@ -0,0 +1,294 @@
+import org.apache.arrow.vector.util.TransferPair;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.plugin.bigquery.BigQueryUtil.toBigQueryColumnName;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.Decimals.encodeShortScaledValue;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.LongTimestampWithTimeZone.fromEpochMillisAndFraction;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MICROS;
+import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_MILLISECOND;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.Math.floorDiv;
+import static java.lang.Math.floorMod;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.apache.arrow.vector.complex.BaseRepeatedValueVector.OFFSET_WIDTH;
+import static org.apache.arrow.vector.types.Types.MinorType.DECIMAL256;
+import static org.apache.arrow.vector.types.Types.MinorType.LIST;
+import static org.apache.arrow.vector.types.Types.MinorType.STRUCT;
+
+public class BigQueryArrowToPageConverter
+        implements AutoCloseable
+{
+    private final VectorSchemaRoot root;
+    private final VectorLoader loader;
+    private final BufferAllocator allocator;
+    private final List<Type> columnTypes;
+    private final List<String> columnNames;
+
+    public BigQueryArrowToPageConverter(BufferAllocator allocator, Schema schema, List<BigQueryColumnHandle> columns)
+    {
+        this.allocator = requireNonNull(allocator, "allocator is null");
+        this.columnTypes = requireNonNull(columns, "columns is null")
+                .stream()
+                .map(BigQueryColumnHandle::getTrinoType)
+                .collect(toImmutableList());
+        this.columnNames = columns.stream()
+                .map(BigQueryColumnHandle::getName)
+                .collect(toImmutableList());
+        List<FieldVector> vectors = schema.getFields()
+                .stream()
+                .map(field -> field.createVector(allocator))
+                .collect(toImmutableList());
+        root = new VectorSchemaRoot(vectors);
+        verify(vectors.size() == columns.size(), "Vectors, columns size differ");
+        loader = new VectorLoader(root, new CommonsCompressionFactory());
```

Comment:
Can we replace `new CommonsCompressionFactory()` with `CommonsCompressionFactory.INSTANCE`?